ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveJoinPushTransitivePredicatesRule.java

```diff
@@ -65,21 +66,15 @@
  */
 public class HiveJoinPushTransitivePredicatesRule extends RelOptRule {
 
-  public static final HiveJoinPushTransitivePredicatesRule INSTANCE_JOIN =
-          new HiveJoinPushTransitivePredicatesRule(HiveJoin.class, HiveRelFactories.HIVE_FILTER_FACTORY);
-
-  public static final HiveJoinPushTransitivePredicatesRule INSTANCE_SEMIJOIN =
-          new HiveJoinPushTransitivePredicatesRule(HiveSemiJoin.class, HiveRelFactories.HIVE_FILTER_FACTORY);
-
-  public static final HiveJoinPushTransitivePredicatesRule INSTANCE_ANTIJOIN =
-          new HiveJoinPushTransitivePredicatesRule(HiveAntiJoin.class, HiveRelFactories.HIVE_FILTER_FACTORY);
-
+  private final HiveConf conf;
   private final FilterFactory filterFactory;
+  private final UnsafeOperatorsFinder unsafeOperatorsFinder = new UnsafeOperatorsFinder();
 
   public HiveJoinPushTransitivePredicatesRule(Class<? extends Join> clazz,
-      FilterFactory filterFactory) {
+      FilterFactory filterFactory, HiveConf conf) {
```

Comment:
I think the rule is only meant to work with `HiveFilter` so I guess it is safe to remove this parameterization from the constructor. I wouldn't consider CBO rules as public API that should be consumed by clients.